x-pack/plugins/endpoint/public/applications/endpoint/store/index.ts

```diff
@@ -4,25 +4,61 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 
-import { createStore, compose, applyMiddleware, Store } from 'redux';
+import {
+  createStore,
+  compose,
+  applyMiddleware,
+  Store,
+  Middleware,
+  Dispatch,
+  MiddlewareAPI,
+} from 'redux';
 import { CoreStart } from 'kibana/public';
 import { appSagaFactory } from './saga';
 import { appReducer } from './reducer';
 import { alertMiddlewareFactory } from './alerts/middleware';
+import { GlobalState } from '../types';
+import { AppAction } from './action';
 
 const composeWithReduxDevTools = (window as any).__REDUX_DEVTOOLS_EXTENSION_COMPOSE__
   ? (window as any).__REDUX_DEVTOOLS_EXTENSION_COMPOSE__({ name: 'EndpointApp' })
   : compose;
 
+export type Selector<S, R> = (state: S) => R;
+
+/**
+ * Wrap Redux Middleware and adjust `getState()` to return the namespace from `GlobalState`
+ * that applies to the given Middleware concern.
+ *
+ * @param selector
+ * @param middleware
+ */
+export const substateMiddlewareFactory = <Substate>(
+  selector: Selector<GlobalState, Substate>,
+  middleware: Middleware<{}, Substate, Dispatch<AppAction>>
+): Middleware<{}, GlobalState, Dispatch<AppAction>> => {
+  return api => {
+    const substateAPI: MiddlewareAPI<Dispatch<AppAction>, Substate> = {
+      ...api,
+      getState() {
+        return selector(api.getState());
+      },
+    };
+    return middleware(substateAPI);
+  };
+};
+
 export const appStoreFactory = (coreStart: CoreStart): [Store, () => void] => {
   const sagaReduxMiddleware = appSagaFactory(coreStart);
   const store = createStore(
     appReducer,
     composeWithReduxDevTools(
-      applyMiddleware(alertMiddlewareFactory(coreStart), appSagaFactory(coreStart))
+      applyMiddleware(
+        substateMiddlewareFactory(s => s.alertList, alertMiddlewareFactory(coreStart).middleware),
+        appSagaFactory(coreStart)
```

Comment:
Just FYI. This is actually a bug @parkiino fixed at one point (should just be the `sagaReduxMiddleware` variable that was created on line 52). That being said, its also going away since Candace refactored it to use Middleware and remove Sagas